### PR TITLE
Automatically install docker-machine-driver-hyperkit if missing or incompatible

### PR DESCRIPTION
**Pull request by josedonizetti**
Fixes https://github.com/kubernetes/minikube/issues/3975

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -1028,62 +1028,27 @@ func saveConfig(clusterCfg *cfg.Config) error {
...
 
-	cmd := exec.Command(driverExecutable, "version")
-	output, err := cmd.Output()
-
-	// we don't want to fail if an error was returned,
-	// libmachine has a nice message for the user if the driver isn't present
+	minikubeVersion, err := version.GetSemverVersion()
 	if err != nil {
-		out.WarningT("Error checking driver version: {{.error}}", out.V{"error": err})
+		out.WarningT("Error parsing minukube version: {{.error}}", out.V{"error": err})
```
Comment (afbjorklund) [inline cmd/minikube/cmd/start.go]: typo, "minikube"

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -1028,62 +1028,27 @@ func saveConfig(clusterCfg *cfg.Config) error {
...
-		targetDir := constants.MakeMiniPath("bin")
-		err := drivers.InstallOrUpdate(driverExecutable, targetDir, minikubeVersion)
-		if err != nil {
-			out.WarningT("Error downloading driver: {{.error}}", out.V{"error": err})
-		}
-		return
 	case constants.DriverHyperkit:
 		driverExecutable = fmt.Sprintf("docker-machine-driver-%s", constants.DriverHyperkit)
-	default: // driver doesn't support version
+	default: // driver doesn't installOrUpdate
```
Comment (afbjorklund) [inline cmd/minikube/cmd/start.go]: spaces ? "install or update"

File: cmd/minikube/cmd/start.go (hunk 1 above)
Comment (afbjorklund) [reply cmd/minikube/cmd/start.go]: I see that this was there since before, though

**Comment by tstromberg** [comment]
Looks great. Just some flakes:

- none: https://github.com/kubernetes/minikube/issues/5332
- hyperkit: https://github.com/kubernetes/minikube/issues/5368

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by mjgallag** [comment]
@tstromberg would it make to follow this same approach for `hyperkit`?

**Pull request merged**